### DISCUSSION TITLE
Run agent on Sysnology DSM

### DISCUSSION THREAD

**Discussion by wolfgangyu** [syagent/agent-2#28] [Ideas]
I really like the syAgent service, so I decided to use it on my Synology NAS. Since Synology DSM already has a GUI scheduling program, I cannot directly use install.sh. Therefore, I manually installed it by following these steps:
1. First, create the etc/syAgent folder.
2. Change the ownership of this folder to the intended user and modify the permissions:
```sudo chown -R user:group /etc/syAgent && sudo chmod -R 700 /etc/syAgent```
```sudo chmod +s 'type -p ping'```
3. Then, put the token in ```/etc/syAgent/sa-auth.log``` and execute ```sh-ageng.sh```.

As a result, the following message appears:
>./sh-agent.sh: line 192: 15887 Killed timeout -s SIGKILL 30 wget -q -o /dev/null -O /etc/syAgent/sh-agent.log -T 25 --post-data "$multipart_data" --no-check-certificate "https://agent.syagent.com/agent"

Can you please advise on how to troubleshoot this issue?"

**Comment by Rashe** [comment]
Hi, it's strange it have timeout. Please check if you have a filrewall, or ports issue. I will try to check on my side too

**Reply by wolfgangyu** [reply]
Currently, it appears that the data has been uploaded, although I still see timeout errors in the log. The two NAS devices I am monitoring are both behind a router and not placed in the DMZ.
